### PR TITLE
Add HTTP options to Keycloak CR

### LINKED ISSUE #14396: Add HTTP options to Keycloak CR
Mapped options:
* http-port
* https-port
* http-enabled
* https-certificate-file
* https-certificate-key-file
* https-trust-store-file (separate issue: https://github.com/keycloak/keycloak/issues/9784)
* https-trust-store-password (separate issue: https://github.com/keycloak/keycloak/issues/9784)
* https-trust-store-type (separate issue: https://github.com/keycloak/keycloak/issues/9784)

Omitting for now for simplicity:
* http-host (probably doesn't make too much sense in K8s)
* http-relative-path
* https-cipher-suites
* https-protocols
* https-client-auth
* https-key-store-file
* https-key-store-password
* https-key-store-type

Removed/moved existing properties in the CR:
* tlsSecret (moved to http/tlsSecret)
* Remove the [`INSECURE-DISABLE` functionality](https://www.keycloak.org/operator/advanced-configuration#_disabling_required_cr_fields) to align with the distribution

CR structure:
```yaml
apiVersion: k8s.keycloak.org/v2alpha1
kind: Keycloak
metadata:
  name: example-kc
spec:
  http:
    httpPort: 80 # integer; reflected by Ingress and Service
    httpsPort: 443 # integer; reflected by Ingress and Service
    tlsSecret: my-super-secret # mapped to https-certificate-file and https-certificate-key-file
...
```

**NOTE:** this is a breaking change and there will be no automated migration; make sure to mention it in release notes and migration guide.

### PR DESCRIPTION
**Pull request by vmuzikar**
Closes #14396.

### CODE REVIEW

**Comment by vmuzikar** [comment]
jFTR – docs will be updated separately in https://github.com/keycloak/keycloak/issues/14924.

This PR is ready for review.

**Review by mabartos — APPROVED** [review]
@vmuzikar Thanks! LGTM!

File: operator/src/main/java/org/keycloak/operator/Constants.java [hunk 1]
```diff
@@ -42,7 +42,8 @@ public final class Constants {
     public static final Map<String, String> DEFAULT_DIST_CONFIG = Map.of(
         "health-enabled","true",
         "cache", "ispn",
-        "cache-stack", "kubernetes"
+        "cache-stack", "kubernetes",
+        "proxy", "passthrough"
```
Comment (mabartos) [inline operator/src/main/java/org/keycloak/operator/Constants.java:46]: @vmuzikar One thing. What's the intention behind this? Is there any documentation for it?

Comment (mabartos) [reply operator/src/main/java/org/keycloak/operator/Constants.java:46]: @vmuzikar Thanks!

**Comment by Pepo48** [comment]
@vmuzikar just a detail but anyway - I noticed that CRDs located in the project root's `kubernetes` aren't regenerated automatically. Therefore, in my PR I changed it manually. What's the procedure here though - Is it supposed to happen automatically or should we update everything at once after a bigger portion of changes or everyone is responsible for updating it on his/her own? (I think the same would apply even to Martin's recent PR).

Thanks!

**Comment by vmuzikar** [comment]
@Pepo48 Thanks for the review! You mean the `target/kubernetes` folder? That should be generated automatically at Maven's `package` phase.

**Comment by Pepo48** [comment]
> @Pepo48 Thanks for the review! You mean the `target/kubernetes` folder? That should be generated automatically at Maven's `package` phase.

nope, I mean this folder: https://github.com/keycloak/keycloak/tree/main/kubernetes

**Comment by vmuzikar** [comment]
@Pepo48 Ah, good point. TBH, I didn't even know about this folder's existence. 😄 Not sure what's its purpose. Maybe @pedroigor might know? In any case, [this](https://github.com/keycloak/keycloak-k8s-resources/tree/nightly) is where the generated CRDs live, we should not be storing any generated files in the main repo.

**Comment by pedroigor** [comment]
@vmuzikar @Pepo48 That is a mistake, I'm sending a PR to remove it.

**Review by pedroigor — APPROVED** [review]

**Pull request merged**